File: src/main/java/at/irian/webstack/middle/bo/GroupFacade.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

package at.irian.webstack.middle.bo;

import at.irian.webstack.middle.orm.security.Group;
import at.irian.webstack.support.cdi.logging.Logger;
import com.avaje.ebean.EbeanServer;
import com.avaje.ebean.annotation.Transactional;

import javax.enterprise.context.Dependent;
import javax.inject.Inject;
import javax.inject.Named;
import java.io.Serializable;

/**
 * @author Werner Punz (latest modification by $Author$)
 * @version $Revision$ $Date$
 */

@Named
@Dependent
public class GroupFacade extends FacadeBase<Group> implements Serializable {

    @Inject
    EbeanServer em;
    @Inject
    Logger log;

    public Group createGroup() {
        return new Group();
    }

    @Transactional
    public void deleteGroup(Group group) {
        if (group.getId() != null) {
            em.delete(group);
        }
    }

}

File: web/src/main/scala/org/extrasapache/myfaces/codi/examples/ebean/business/bo/security/GroupFacade.scala
package org.extrasapache.myfaces.codi.examples.ebean.business.bo.security

import org.extrasapache.myfaces.codi.examples.ebean.business.bo.common.FacadeBase
import org.extrasapache.myfaces.codi.examples.ebean.orm.security.SecGroup

import java.util._

import collection.mutable.ArrayBuffer
import com.avaje.ebean.annotation.Transactional
import collection.JavaConversions._
import javax.enterprise.context.Dependent
import javax.inject.Named

/**
 *
 * @author Werner Punz (latest modification by $Author$)
 * @version $Revision$ $Date$
 */

@Named
@Dependent
@serializable
class GroupFacade extends FacadeBase[SecGroup] {
  clazz = classOf[SecGroup]

  def createGroup: SecGroup = new SecGroup
  import java.lang.Long._



  def loadByIdsStr(ids: Collection[String]): List[SecGroup] = {
    /*we use our implicit conversion to map the lists*/
    implicit def mapCollections(ids: Collection[String]): Collection[Long] = {
      asScalaIterable(ids).map{parseLong(_)}
    }

    loadByIds(ids)
  }

  def loadByIds(ids: Collection[Long]): List[SecGroup] = {
    val query = em.createQuery(clazz)
    query.where().in("id", ids)
    query.findList
  }

  @Transactional
  def saveAll(groups: Collection[SecGroup]) {
    for (group <- asScalaIterable(groups)) em.save(group)
  }

  def deleteGroup(group: SecGroup) = delete(group)



}